e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -0,0 +1,154 @@
+
+type MyFixtures = {
+  portal: Portal;
+};
+
+export const test = baseTest.extend<MyFixtures>({
+  portal: async ({ page }, use) => {
+    const portal = new Portal(page);
+    await portal.doCredentialsLogin();
+    await use(portal);
+  },
+});
+
+async function retryForCondition(condition: Promise<any>, retries = 10, retryInterval = 500) {
+  while (retries) {
+    if (await condition) {
+      break;
+    } else {
+      setTimeout(() => {
+        retries -= 1;
+      }, retryInterval);
+    }
+  }
+}
+
+async function getRowByTenantProduct(
+  page: Page,
+  tenant: string,
+  product: string,
+  tenantProductIndexes: Array<number>
+): Promise<Locator | string> {
+  const rows = await page.getByRole('row').all();
+  const setupLinks = await Promise.all(rows.map(async (el: Locator) => el));
+  let row: Locator | string = '';
+
+  for (const link of setupLinks) {
+    const el = await link.innerText();
+    const cellContents = el.split('\t');
+    if (
+      cellContents.length > 1 &&
+      cellContents[tenantProductIndexes[0]] === tenant &&
+      cellContents[tenantProductIndexes[1]] === product
+    ) {
+      row = link;
+    }
+  }
+
+  return row;
+}
+
+test.describe('Admin Portal Enterprise SSO SetupLink using generic SAML 2.0', () => {
+  test('should be able to create setup link and sso connection using generic SAML 2.0', async ({ page }) => {
+    const tenant = 'acme-setuplink-test.com';
+    const product = 'acme-setuplink-test';
+
+    // Go to admin/sso-connection/setup-link page and create setup link
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+    await page.getByRole('button', { name: 'New Setup Link' }).click();
+    await page.getByPlaceholder('Acme SSO').fill('acme-test');
+    await page.getByLabel('Description (Optional)').fill('acme test');
+    await page.getByPlaceholder('acme', { exact: true }).fill(tenant);
+    await page.getByPlaceholder('MyApp').fill(product);
+    await page.getByPlaceholder('http://localhost:3366', { exact: true }).fill(TEST_SETUPLINK_REDIRECT_URL);
+    await page.getByPlaceholder('http://localhost:3366/login/').fill(TEST_SETUPLINK_DEFAULT_REDIRECT_URL);
+
+    await page.getByRole('button', { name: 'Create Setup Link' }).click();
+
+    // Extract generated setup link
+    const linkContent = await page
+      .getByText(TEST_SETUPLINK_URL_LABEL_SELECTOR)
+      .locator('input[type="text"]')
+      .first()
+      .inputValue();
+
+    // Go back to new connections page
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+
+    // Await for rows loaded
+    await expect(page.getByRole('table')).toBeVisible();
+
+    // Get single row containing setuplink
+    let linkRow = await getRowByTenantProduct(page, tenant, product, [0, 1]);
+    expect(linkRow, 'Failed to create setup link').toBeTruthy();
+
+    // Open new tab and go to setup link page
+    const context = page.context();
+    const page1 = await context.newPage();
+    await page1.waitForLoadState();
```

Comment:
Not needed here, 
- Page has not navigated yet
- [Also](https://playwright.dev/docs/api/class-page#page-wait-for-load-state) 
  > Most of the time, this method is not needed because Playwright [auto-waits before every action](https://playwright.dev/docs/actionability).